Add concept map documentation

neenjaw opened the pull request:
Recent comments on the forum bring to light that there is relatively sparse documentation on:
- the requirements that were determined for the concept map
- the method of computation determining the specification for the concept map
- details about the resulting graph

neenjaw committed:
add concept map docs

ErikSchierboom commented on building/tracks/concept-map.md:
```diff
@@ -0,0 +1,48 @@
+
+- Provide a meaningful concept map of ideas from simple to complex ideas.
+- Provide a pathway towards fluency in a language.
+- Illustrate the progression through the course [track] contents.
```
What does `[track]` refer to?

neenjaw replied:
I had added that as a colloquialism, but probably isn't necessary

neenjaw replied:
```suggestion
- Illustrate the progression through the course contents.
```

ErikSchierboom commented on building/tracks/concept-map.md:
````diff
@@ -0,0 +1,48 @@
+    linkStyle 2 stroke-width:2px,fill:none,stroke:red;
+```
+
+[github-concept-code]: https://github.com/exercism/website/blob/e47525f66b036662138e5e41055430fe7963a282/app/commands/track/determine_concept_map_layout.rb
````
Maybe point to the version that is in `main`?

```suggestion
[github-concept-code]: https://github.com/exercism/website/blob/main/app/commands/track/determine_concept_map_layout.rb
```

ErikSchierboom commented on building/tracks/concept-map.md:
```diff
@@ -0,0 +1,48 @@
+
+## Configuration
+
+The configuration of the concept map is determined by details of the `config.json` associated to a track.
```
```suggestion
The configuration of the concept map is determined by details of the [`config.json`](/docs/building/tracks/config-json) associated to a track.
```

But then as a reference link

ErikSchierboom reviewed (commented):
Couple of tiny comments:

- Could you add an entry to https://github.com/exercism/docs/blob/main/building/config.json
- Could you refer to the document from the "Concepts" section here: https://github.com/exercism/docs/blob/main/building/tracks/README.md#concepts
- Maybe also add a link to this document from https://github.com/exercism/docs/blob/main/building/tracks/concepts.md?

neenjaw committed:
Update building/tracks/concept-map.md

Co-authored-by: Erik Schierboom <erik_schierboom@hotmail.com>

neenjaw committed:
Update building/tracks/concept-map.md

Co-authored-by: Erik Schierboom <erik_schierboom@hotmail.com>

neenjaw committed:
Update building/tracks/concept-map.md

ErikSchierboom commented:
Great. All that's left to do is:

> Could you add an entry to https://github.com/exercism/docs/blob/main/building/config.json
> Could you refer to the document from the "Concepts" section here: https://github.com/exercism/docs/blob/main/building/tracks/README.md#concepts
> Maybe also add a link to this document from https://github.com/exercism/docs/blob/main/building/tracks/concepts.md?

neenjaw commented:
@ErikSchierboom Yea, I didn't have time to address those points yet.  Will re-request review when completed. 😄

neenjaw committed:
add config and links

Add configuration for the concept map config
Add links from the existing building track README and concept
documentation.

neenjaw commented:
I think I have those other areas covered now, @ErikSchierboom 😄

ErikSchierboom reviewed (approved):
Excellent!
